nifi-nar-bundles/nifi-extension-utils/nifi-processor-utils/src/main/java/org/apache/nifi/processor/util/listen/dispatcher/SocketChannelDispatcher.java

```diff
@@ -52,61 +53,80 @@
+                                   final ByteBufferSource bufferSource,
                                    final BlockingQueue<E> events,
                                    final ComponentLog logger,
                                    final int maxConnections,
                                    final SSLContext sslContext,
+                                   final ClientAuth clientAuth,
                                    final Charset charset) {
-        this(eventFactory, handlerFactory, bufferPool, events, logger, maxConnections, sslContext, ClientAuth.REQUIRED, charset);
+        this(eventFactory, handlerFactory, bufferSource, events, logger, maxConnections, maxConnections, sslContext, clientAuth, charset);
     }
 
     public SocketChannelDispatcher(final EventFactory<E> eventFactory,
                                    final ChannelHandlerFactory<E, AsyncChannelDispatcher> handlerFactory,
-                                   final BlockingQueue<ByteBuffer> bufferPool,
+                                   final ByteBufferSource bufferSource,
                                    final BlockingQueue<E> events,
                                    final ComponentLog logger,
                                    final int maxConnections,
+                                   final int maxThreadPoolSize,
                                    final SSLContext sslContext,
                                    final ClientAuth clientAuth,
                                    final Charset charset) {
         this.eventFactory = eventFactory;
         this.handlerFactory = handlerFactory;
-        this.bufferPool = bufferPool;
+        this.bufferSource = bufferSource;
         this.events = events;
         this.logger = logger;
         this.maxConnections = maxConnections;
+        this.maxThreadPoolSize = maxThreadPoolSize;
         this.keyQueue = new LinkedBlockingQueue<>(maxConnections);
         this.sslContext = sslContext;
         this.clientAuth = clientAuth;
         this.charset = charset;
-
-        if (bufferPool == null || bufferPool.size() == 0 || bufferPool.size() != maxConnections) {
-            throw new IllegalArgumentException(
-                    "A pool of available ByteBuffers equal to the maximum number of connections is required");
-        }
     }
 
     @Override
     public void open(final InetAddress nicAddress, final int port, final int maxBufferSize) throws IOException {
+        final InetSocketAddress inetSocketAddress = new InetSocketAddress(nicAddress, port);
+
         stopped = false;
-        executor = Executors.newFixedThreadPool(maxConnections);
+        executor = new ThreadPoolExecutor(
+                maxThreadPoolSize,
+                maxThreadPoolSize,
+                60L,
+                TimeUnit.SECONDS,
+                new LinkedBlockingQueue<>(),
+                new BasicThreadFactory.Builder().namingPattern(inetSocketAddress.toString() + "-dispatcher-%d").build());
```

Comment:
These are "worker" threads started by the dispatcher so "-worker" would be the appropriate suffix.

Regarding the prefix: It is more typical to use the processor's (class)name + id like in case of the dispatcher thread: https://github.com/apache/nifi/blob/fe950131c35756dabd677fb21b436a1f85eabced/nifi-nar-bundles/nifi-extension-utils/nifi-processor-utils/src/main/java/org/apache/nifi/processor/util/listen/AbstractListenEventProcessor.java#L198

It would be better to link the dispatcher and the worker threads together via a same naming convention (eg. to make troubleshooting easier).
I see that the processor's attributes are not available here right now so this change would not be straightforward.
I would suggest reverting it back to the original logic (default pool-x-thread-y) and solving it when the other listen processors are getting fixed.

Reply:
Yes, the decision was made due to the processor attributes are not available here. Also, I decided to change the default value to make it easier to follow the threads in case multiple ListenTCP processors are in the flow. By adding the address, it is easy to identify which processor it is related to which adding the class does not help. I am aware of that this is not following the general convention but due to the possible multiplication of the pools it makes sense to me. Adding the class as prefix as well might work but it might be too long. For now I went with a common ground type of solution and replaced dispatcher postfix with worker. For the prefix I intend to keep it, expect you insist to change.

Reply:
It is fine with me. Though the network address is not the best but much better than pool-x-thread-y.